### ISSUE TITLE
[-Wsign-compare] warning in tests\catch.hpp lines 844 & 874

### ISSUE THREAD

**Issue by SteeleDynamics** [gabime/spdlog#483]
Comparison of signed and unsigned integers warning. (MinGW-64, gcc 7.1.0)

**Comment by vedranmiletic** [comment]
Which version has the fix? I still see this in 1.3.1.

**Comment by vedranmiletic** [comment]
Here's a bit more details. I don't get it on 32-bit MinGW. On 64-bit MinGW I get:

```
Compiling C++ object 'subprojects/spdlog-1.3.1/tests/7c1051b@@tests_exe@exe/test_file_helper.cpp.obj'.                             
In file included from ../subprojects/spdlog-1.3.1/tests/includes.h:3,                                                                      
                 from ../subprojects/spdlog-1.3.1/tests/test_file_helper.cpp:4:                                                            
../subprojects/spdlog-1.3.1/tests/catch.hpp: In instantiation of 'static bool Catch::Internal::Evaluator<T1, T2, (Catch::Internal::Operator)
0>::evaluate(const T1&, const T2&) [with T1 = long long unsigned int; T2 = int]':                                                          
../subprojects/spdlog-1.3.1/tests/catch.hpp:889:47:   required from 'bool Catch::Internal::compare(const T1&, const T2&) [with Catch::Intern
al::Operator Op = (Catch::Internal::Operator)0; T1 = long long unsigned int; T2 = int]'                                                    
../subprojects/spdlog-1.3.1/tests/catch.hpp:1391:50:   required from 'Catch::ResultBuilder& Catch::ExpressionLhs<T>::captureExpression(const
 RhsT&) [with Catch::Internal::Operator Op = (Catch::Internal::Operator)0; RhsT = int; T = const long long unsigned int&]'                 
../subprojects/spdlog-1.3.1/tests/catch.hpp:1334:54:   required from 'Catch::ResultBuilder& Catch::ExpressionLhs<T>::operator==(const RhsT&)
 [with RhsT = int; T = const long long unsigned int&]'                                                                                     
../subprojects/spdlog-1.3.1/tests/test_file_helper.cpp:56:5:   required from here                                                          
../subprojects/spdlog-1.3.1/tests/catch.hpp:844:34: warning: comparison of integer expressions of different signedness: 'long long unsigned 
int' and 'int' [-Wsign-compare]                                                                                                            
             return opCast( lhs ) ==  opCast( rhs );                                                                                       
                    ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
```
